cdk/domino_cdk/provisioners/eks.py

```diff
@@ -0,0 +1,518 @@
+from os.path import isfile
+from re import MULTILINE
+from re import split as re_split
+from typing import Any, List, Optional, Tuple, Type, Union
+
+import aws_cdk.aws_ec2 as ec2
+import aws_cdk.aws_eks as eks
+import aws_cdk.aws_iam as iam
+import aws_cdk.aws_lambda as aws_lambda
+from aws_cdk import aws_autoscaling
+from aws_cdk import core as cdk
+from aws_cdk.lambda_layer_awscli import AwsCliLayer
+from aws_cdk.lambda_layer_kubectl import KubectlLayer
+from requests import get as requests_get
+from yaml import safe_load as yaml_safe_load
+
+from domino_cdk.config import EKS, MachineImage
+
+manifests = [
+    [
+        "calico",
+        "https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/v1.7.10/config/master/calico.yaml",
+    ]
+]
+
+
+class DominoEksProvisioner:
```

Comment:
May be rename EKS to Cluster in this file?

Here we have a mix of EKS, IAM, EC2, IAM/S3 and IAM/ECR. Plus bastion and Calico.

Reply:
Tell me what you think of the latest commit. I pulled Calico out into its own thing, and made the cluster, iam provisioning and nodegroup provisioning their own submodules where it's more distinct what role they each play in provisioning the eks cluster and how they each linearly input into each other ("provision the control plane" -> "create the IAM role for the nodegroups" -> "provision the worker nodes").

Reply:
I like it